### PR TITLE
ENG-4991 add health check ingress

### PR DESCRIPTION
**Pull request by avdev4j**

### CODE REVIEW

File: src/models/bundle-descriptor-constraints.ts [hunk 1]
```diff
@@ -242,6 +242,10 @@ export const MICROSERVICE_CONSTRAINTS: ObjectConstraints<Microservice> = {
     required: true,
     type: 'string'
   },
+  healthCheckIngress: {
+    required: true,
```
Comment (antromeo) [inline src/models/bundle-descriptor-constraints.ts]: Should the field healthCheckIngress be required? I think no for backwards compatibility.
In this case, in old bundles, any commands you execute returns the following error. 
```
Error: entando.json is not valid.
Field "healthCheckIngress" is required
```

Furthermore, if you want to bind the values to the two enumeratives, you should add:
```
validators: [values(HealthCheckIngress)]
```

File: src/services/microservice-service.ts [hunk 2]
```diff
@@ -55,7 +55,8 @@ export class MicroserviceService {
 
     this.addMicroserviceDescriptor({
       ...ms,
-      healthCheckPath: ms.healthCheckPath ?? DEFAULT_HEALTH_CHECK_PATH
+      healthCheckPath: ms.healthCheckPath ?? DEFAULT_HEALTH_CHECK_PATH,
+      healthCheckIngress: ms.healthCheckIngress ?? HealthCheckIngress.Canonical
```
Comment (antromeo) [inline src/services/microservice-service.ts:58]: ok for input but in output when you run `ent bundle pack` you should find your property in output (so`.entando/output/plugins/plugin-desc.yaml`) in that way the CM can processing the CR when it is applied.
try and let me know

File: src/models/bundle-descriptor-constraints.ts (hunk 1 above)
Comment (avdev4j) [reply src/models/bundle-descriptor-constraints.ts]: @antromeo I checked the ticket and it is not defined if the property should be required or not. So, we can update that!

File: src/services/microservice-service.ts (hunk 2 above)
Comment (avdev4j) [reply src/services/microservice-service.ts:58]: let's discuss that later. Not sure to get it

Comment (avdev4j) [reply src/services/microservice-service.ts:58]: done

File: src/models/bundle-descriptor-constraints.ts (hunk 1 above)
Comment (avdev4j) [reply src/models/bundle-descriptor-constraints.ts]: done

**Review by antromeo — APPROVED** [review]

**Pull request merged**